### PR TITLE
fix(cf): make start on creation flag work again

### PR DESCRIPTION
**Pull request by jmelchio**
- spinnaker/spinnaker#3992

### CODE REVIEW

File: clouddriver-cloudfoundry/src/test/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/ops/DeployCloudFoundryServerGroupAtomicOperationTest.java [hunk 1]
```diff
@@ -128,6 +129,142 @@ void executeOperation() {
...
+    inOrder.verify(cloudFoundryClient.getServiceInstances()).createServiceBindingsByName(any(), eq(Collections.singletonList("service1")));
+    inOrder.verify(apps).startApplication("serverGroupId");
+
+    assertThat(testTask.getStatus().isFailed()).isTrue();
+  }
+
+  @Test
+  void executeOperationWithNoStartFlag() {
+    // Given
+    final DeployCloudFoundryServerGroupDescription description = new DeployCloudFoundryServerGroupDescription()
```
Comment (jkschneider) [inline clouddriver-cloudfoundry/src/test/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/ops/DeployCloudFoundryServerGroupAtomicOperationTest.java]: Are `DeployCloudFoundryServerGroupDescription` in this and prior test the same except for a property or two? Can they be extracted out of the method?

Comment (jmelchio) [reply clouddriver-cloudfoundry/src/test/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/ops/DeployCloudFoundryServerGroupAtomicOperationTest.java]: Yes, we can probably extract that. Will fix in the morning.

Comment (jmelchio) [reply clouddriver-cloudfoundry/src/test/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/ops/DeployCloudFoundryServerGroupAtomicOperationTest.java]: Extracted code and some other duplicated code.

File: clouddriver-cloudfoundry/src/test/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/ops/DeployCloudFoundryServerGroupAtomicOperationTest.java [hunk 2]
```diff
@@ -66,8 +66,114 @@ void convertToMbHandling() {
...
+      io.vavr.collection.List.of("buildpack1", "buildpack2").asJava(),
+      HashMap.of(
+        "token", "ASDF"
+      ).toJavaMap());
+    inOrder.verify(apps).uploadPackageBits(eq("serverGroupId_package"), any());
+    inOrder.verify(apps).createBuild("serverGroupId_package");
+    inOrder.verify(apps).scaleApplication("serverGroupId", 7, 1024, 2048);
+    inOrder.verify(apps).updateProcess("serverGroupId", null, "http", "/health");
+    inOrder.verify(cloudFoundryClient.getServiceInstances()).createServiceBindingsByName(any(), eq(Collections.singletonList("service1")));
+    inOrder.verify(apps, never()).startApplication("serverGroupId");
```
Comment (claymccoy) [inline clouddriver-cloudfoundry/src/test/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/ops/DeployCloudFoundryServerGroupAtomicOperationTest.java]: This is the important part. It gets lost in all the other verifications which are mostly duplicated between the two tests. Is there a way to set this important verify apart and combine the common ones?

Comment (jmelchio) [reply clouddriver-cloudfoundry/src/test/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/ops/DeployCloudFoundryServerGroupAtomicOperationTest.java]: I'm sure we can figure that out. I'll take a look at it.

Comment (jmelchio) [reply clouddriver-cloudfoundry/src/test/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/ops/DeployCloudFoundryServerGroupAtomicOperationTest.java]: I think this solves the issue, @claymccoy please take a look.

**Pull request merged**